src/cleaning/alru.c

```diff
@@ -517,7 +517,7 @@ static int ocf_alru_recovery_handle(ocf_parallelize_t parallelize,
 	int i;
 
 	portion = DIV_ROUND_UP((uint64_t)entries, shards_cnt);
-	begin = portion*shard_id;
+	begin = (uint64_t)portion*shard_id;
```

Comment:
I would approach it differently. I'm not sure why begin and end are 64 bit in the first place. I would change then to 32bit unsigned.